Lib/json/__init__.py

```diff
@@ -342,6 +343,14 @@ def loads(s, *, encoding=None, cls=None, object_hook=None, parse_float=None,
                             f'not {s.__class__.__name__}')
         s = s.decode(detect_encoding(s), 'surrogatepass')
 
+    if encoding is not _sentinel:
```

Comment:
How about detecting `encoding` with `kw`?

Remove `encoding` in arguments and:

```py
if 'encoding' in kw:
    del kw['encoding']
    # raise warning
```

Reply:
That's actually way better indeed.